### ISSUE TITLE
cocos gen-libs -p android error

### ISSUE THREAD

**Issue by deguilardi** [cocos2d/cocos2d-x#18668]
- cocos2d-x version: 3.17
- devices test on: N/A
- developing environments
   - NDK version: 13b
   - Xcode version: 9.2
   - VS version: N/A
   - browser type and version: N/A

Steps to Reproduce:

1. run command cocos gen-libs -p android --app-abi armeabi-v7a

LOG:

Android NDK: WARNING: APP_PLATFORM android-16 is larger than android:minSdkVersion 9 in ./AndroidManifest.xml    
[armeabi-v7a] Compile++ arm  : cocos2d_lua_static <= lua_cocos2dx_extension_auto.cpp
[armeabi-v7a] StaticLibrary  : libsimulator.a
[armeabi-v7a] Compile++ arm  : cocos2d_js_static <= jsb_cocos2dx_extension_auto.cpp
[armeabi-v7a] Compile++ arm  : cocos2d_js_static <= js_manual_conversions.cpp
[armeabi-v7a] StaticLibrary  : libcocos2d.a
[armeabi-v7a] StaticLibrary  : liblocalstorage.a
[armeabi-v7a] Compile++ arm  : cocostudio_static <= CCActionTimeline.cpp
[armeabi-v7a] Compile++ arm  : cocostudio_static <= CCActionTimelineNode.cpp
[armeabi-v7a] Compile++ arm  : cocostudio_static <= CSLoader.cpp
[armeabi-v7a] Compile++ arm  : cocostudio_static <= CCBoneNode.cpp
[armeabi-v7a] Compile++ arm  : cocostudio_static <= CCSkeletonNode.cpp
[armeabi-v7a] Compile++ arm  : cocostudio_static <= CCSkinNode.cpp
[armeabi-v7a] Compile++ arm  : cocostudio_static <= ArmatureNodeReader.cpp
[armeabi-v7a] Compile++ arm  : cocostudio_static <= TabControlReader.cpp
[armeabi-v7a] Compile++ arm  : cocostudio_static <= CocoStudio.cpp
[armeabi-v7a] Compile++ arm  : cocos_ui_static <= UILayoutComponent.cpp
[armeabi-v7a] Compile++ arm  : cocos_ui_static <= UITabControl.cpp
[armeabi-v7a] Compile++ thumb: audioengine_static <= CCThreadPool.cpp
[armeabi-v7a] Compile++ thumb: audioengine_static <= AssetFd.cpp
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp: In function 'jsval asset_to_jsval(JSContext*, const cocos2d::extension::ManifestAsset&)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp:2929:94: error: invalid use of incomplete type 'const struct cocos2d::extension::ManifestAsset'
     bool ok = JS_DefineProperty(cx, tmp, "md5", JS::RootedValue(cx, std_string_to_jsval(cx, v.md5)), JSPROP_ENUMERATE | JSPROP_PERMANENT) &&
                                                                                              ^
In file included from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp:25:0:
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../scripting/js-bindings/manual/js_manual_conversions.h:53:12: error: forward declaration of 'const struct cocos2d::extension::ManifestAsset'
     struct ManifestAsset;
            ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp:2930:85: error: invalid use of incomplete type 'const struct cocos2d::extension::ManifestAsset'
     JS_DefineProperty(cx, tmp, "path", JS::RootedValue(cx, std_string_to_jsval(cx, v.path)), JSPROP_ENUMERATE | JSPROP_PERMANENT) &&
                                                                                     ^
In file included from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp:25:0:
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../scripting/js-bindings/manual/js_manual_conversions.h:53:12: error: forward declaration of 'const struct cocos2d::extension::ManifestAsset'
     struct ManifestAsset;
            ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp:2931:47: error: invalid use of incomplete type 'const struct cocos2d::extension::ManifestAsset'
     JS_DefineProperty(cx, tmp, "compressed", v.compressed, JSPROP_ENUMERATE | JSPROP_PERMANENT) &&
                                               ^
In file included from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp:25:0:
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../scripting/js-bindings/manual/js_manual_conversions.h:53:12: error: forward declaration of 'const struct cocos2d::extension::ManifestAsset'
     struct ManifestAsset;
            ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp:2932:41: error: invalid use of incomplete type 'const struct cocos2d::extension::ManifestAsset'
     JS_DefineProperty(cx, tmp, "size", v.size, JSPROP_ENUMERATE | JSPROP_PERMANENT) &&
                                         ^
In file included from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp:25:0:
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../scripting/js-bindings/manual/js_manual_conversions.h:53:12: error: forward declaration of 'const struct cocos2d::extension::ManifestAsset'
     struct ManifestAsset;
            ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp:2933:55: error: invalid use of incomplete type 'const struct cocos2d::extension::ManifestAsset'
     JS_DefineProperty(cx, tmp, "downloadState", (int)v.downloadState, JSPROP_ENUMERATE | JSPROP_PERMANENT);
                                                       ^
In file included from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../manual/js_manual_conversions.cpp:25:0:
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../scripting/js-bindings/manual/js_manual_conversions.h:53:12: error: forward declaration of 'const struct cocos2d::extension::ManifestAsset'
     struct ManifestAsset;
            ^
[armeabi-v7a] Compile++ thumb: audioengine_static <= AudioDecoder.cpp
[armeabi-v7a] Compile++ thumb: audioengine_static <= AudioDecoderProvider.cpp
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp: In function 'int lua_cocos2dx_extension_ScrollView_setSwallowTouches(lua_State*)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp:8475:15: error: 'class cocos2d::extension::ScrollView' has no member named 'setSwallowTouches'
         cobj->setSwallowTouches(arg0);
               ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp: In function 'int lua_cocos2dx_extension_AssetsManagerEx_getMaxConcurrentTask(lua_State*)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp:12658:31: error: 'class cocos2d::extension::AssetsManagerEx' has no member named 'getMaxConcurrentTask'
         const int ret = cobj->getMaxConcurrentTask();
                               ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp: In function 'int lua_cocos2dx_extension_AssetsManagerEx_setVerifyCallback(lua_State*)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp:12747:49: error: 'ManifestAsset' is not a member of 'cocos2d::extension'
         std::function<bool (const std::string&, cocos2d::extension::ManifestAsset)> arg0;
                                                 ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp:12747:82: error: expression list treated as compound expression in functional cast [-fpermissive]
         std::function<bool (const std::string&, cocos2d::extension::ManifestAsset)> arg0;
                                                                                  ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp:12747:83: error: template argument 1 is invalid
         std::function<bool (const std::string&, cocos2d::extension::ManifestAsset)> arg0;
                                                                                   ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp:12747:89: error: invalid type in declaration before ';' token
         std::function<bool (const std::string&, cocos2d::extension::ManifestAsset)> arg0;
                                                                                         ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp:12759:15: error: 'class cocos2d::extension::AssetsManagerEx' has no member named 'setVerifyCallback'
         cobj->setVerifyCallback(arg0);
               ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp: In function 'int lua_cocos2dx_extension_AssetsManagerEx_setVersionCompareHandle(lua_State*)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp:12907:15: error: 'class cocos2d::extension::AssetsManagerEx' has no member named 'setVersionCompareHandle'
         cobj->setVersionCompareHandle(arg0);
               ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp: In function 'int lua_cocos2dx_extension_AssetsManagerEx_setMaxConcurrentTask(lua_State*)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/lua-bindings/proj.android/../auto/lua_cocos2dx_extension_auto.cpp:12957:15: error: 'class cocos2d::extension::AssetsManagerEx' has no member named 'setMaxConcurrentTask'
         cobj->setMaxConcurrentTask(arg0);
               ^
[armeabi-v7a] Compile++ thumb: audioengine_static <= AudioDecoderSLES.cpp
[armeabi-v7a] Compile++ thumb: audioengine_static <= AudioDecoderOgg.cpp
[armeabi-v7a] Compile++ thumb: audioengine_static <= AudioDecoderMp3.cpp
[armeabi-v7a] Compile++ thumb: audioengine_static <= AudioDecoderWav.cpp
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp: In function 'bool js_cocos2dx_extension_ScrollView_setSwallowTouches(JSContext*, uint32_t, jsval*)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:4998:15: error: 'class cocos2d::extension::ScrollView' has no member named 'setSwallowTouches'
         cobj->setSwallowTouches(arg0);
               ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp: In function 'bool js_cocos2dx_extension_AssetsManagerEx_getMaxConcurrentTask(JSContext*, uint32_t, jsval*)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:6699:31: error: 'class cocos2d::extension::AssetsManagerEx' has no member named 'getMaxConcurrentTask'
         const int ret = cobj->getMaxConcurrentTask();
                               ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp: In lambda function:
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:6740:89: error: 'larg1' has incomplete type
           auto lambda = [=](const std::string& larg0, cocos2d::extension::ManifestAsset larg1) -> bool {
                                                                                         ^
In file included from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../scripting/js-bindings/manual/ScriptingCore.h:36:0,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../scripting/js-bindings/manual/cocos2d_specifics.hpp:27,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:2:
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../scripting/js-bindings/manual/js_manual_conversions.h:53:12: error: forward declaration of 'struct cocos2d::extension::ManifestAsset'
     struct ManifestAsset;
            ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp: In function 'bool js_cocos2dx_extension_AssetsManagerEx_setVerifyCallback(JSContext*, uint32_t, jsval*)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:6756:16: error: no match for 'operator=' (operand types are 'std::function<bool(const std::basic_string<char>&, cocos2d::extension::ManifestAsset)>' and 'js_cocos2dx_extension_AssetsManagerEx_setVerifyCallback(JSContext*, uint32_t, jsval*)::<lambda(const string&, cocos2d::extension::ManifestAsset)>')
           arg0 = lambda;
                ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:6756:16: note: candidates are:
In file included from /Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/thread:39:0,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../base/CCConsole.h:42,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../base/ccMacros.h:35,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../2d/CCNode.h:33,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../2d/CCProtectedNode.h:33,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../ui/UIWidget.h:28,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../ui/CocosGUI.h:29,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../scripting/js-bindings/manual/ScriptingCore.h:32,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/3d/../scripting/js-bindings/manual/cocos2d_specifics.hpp:27,
                 from /Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:2:
/Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/functional:2241:7: note: std::function<_Res(_ArgTypes ...)>& std::function<_Res(_ArgTypes ...)>::operator=(const std::function<_Res(_ArgTypes ...)>&) [with _Res = bool; _ArgTypes = {const std::basic_string<char, std::char_traits<char>, std::allocator<char> >&, cocos2d::extension::ManifestAsset}]
       operator=(const function& __x)
       ^
/Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/functional:2241:7: note:   no known conversion for argument 1 from 'js_cocos2dx_extension_AssetsManagerEx_setVerifyCallback(JSContext*, uint32_t, jsval*)::<lambda(const string&, cocos2d::extension::ManifestAsset)>' to 'const std::function<bool(const std::basic_string<char>&, cocos2d::extension::ManifestAsset)>&'
/Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/functional:2259:7: note: std::function<_Res(_ArgTypes ...)>& std::function<_Res(_ArgTypes ...)>::operator=(std::function<_Res(_ArgTypes ...)>&&) [with _Res = bool; _ArgTypes = {const std::basic_string<char, std::char_traits<char>, std::allocator<char> >&, cocos2d::extension::ManifestAsset}]
       operator=(function&& __x)
       ^
/Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/functional:2259:7: note:   no known conversion for argument 1 from 'js_cocos2dx_extension_AssetsManagerEx_setVerifyCallback(JSContext*, uint32_t, jsval*)::<lambda(const string&, cocos2d::extension::ManifestAsset)>' to 'std::function<bool(const std::basic_string<char>&, cocos2d::extension::ManifestAsset)>&&'
/Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/functional:2273:7: note: std::function<_Res(_ArgTypes ...)>& std::function<_Res(_ArgTypes ...)>::operator=(std::nullptr_t) [with _Res = bool; _ArgTypes = {const std::basic_string<char, std::char_traits<char>, std::allocator<char> >&, cocos2d::extension::ManifestAsset}; std::nullptr_t = std::nullptr_t]
       operator=(nullptr_t)
       ^
/Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/functional:2273:7: note:   no known conversion for argument 1 from 'js_cocos2dx_extension_AssetsManagerEx_setVerifyCallback(JSContext*, uint32_t, jsval*)::<lambda(const string&, cocos2d::extension::ManifestAsset)>' to 'std::nullptr_t'
/Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/functional:2302:2: note: template<class _Functor> std::function<_Res(_ArgTypes ...)>::_Requires<std::__and_<std::__not_<std::is_same<typename std::decay<_Tp>::type, std::function<_Res(_ArgTypes ...)> > >, std::__or_<std::is_void<_Tp>, std::is_convertible<std::function<_Res(_ArgTypes ...)>::_Invoke<typename std::decay<_Tp>::type>, _Res> > >, std::function<_Res(_ArgTypes ...)>&> std::function<_Res(_ArgTypes ...)>::operator=(_Functor&&) [with _Functor = _Functor; _Res = bool; _ArgTypes = {const std::basic_string<char, std::char_traits<char>, std::allocator<char> >&, cocos2d::extension::ManifestAsset}]
  operator=(_Functor&& __f)
  ^
/Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/functional:2302:2: note:   template argument deduction/substitution failed:
/Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/functional:2311:2: note: template<class _Functor> std::function<_Res(_ArgTypes ...)>& std::function<_Res(_ArgTypes ...)>::operator=(std::reference_wrapper<_Tp>) [with _Functor = _Functor; _Res = bool; _ArgTypes = {const std::basic_string<char, std::char_traits<char>, std::allocator<char> >&, cocos2d::extension::ManifestAsset}]
  operator=(reference_wrapper<_Functor> __f) noexcept
  ^
/Users/deguilardi/Library/Android/android-ndk-r13b/sources/cxx-stl/gnu-libstdc++/4.9/include/functional:2311:2: note:   template argument deduction/substitution failed:
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:6756:16: note:   'js_cocos2dx_extension_AssetsManagerEx_setVerifyCallback(JSContext*, uint32_t, jsval*)::<lambda(const string&, cocos2d::extension::ManifestAsset)>' is not derived from 'std::reference_wrapper<_Tp>'
           arg0 = lambda;
                ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:6765:15: error: 'class cocos2d::extension::AssetsManagerEx' has no member named 'setVerifyCallback'
         cobj->setVerifyCallback(arg0);
               ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp: In function 'bool js_cocos2dx_extension_AssetsManagerEx_setVersionCompareHandle(JSContext*, uint32_t, jsval*)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:6847:15: error: 'class cocos2d::extension::AssetsManagerEx' has no member named 'setVersionCompareHandle'
         cobj->setVersionCompareHandle(arg0);
               ^
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp: In function 'bool js_cocos2dx_extension_AssetsManagerEx_setMaxConcurrentTask(JSContext*, uint32_t, jsval*)':
/Users/deguilardi/cocosWorkspace/lib-cocos2d/cocos/scripting/js-bindings/proj.android/../auto/jsb_cocos2dx_extension_auto.cpp:6867:15: error: 'class cocos2d::extension::AssetsManagerEx' has no member named 'setMaxConcurrentTask'
         cobj->setMaxConcurrentTask(arg0);
               ^
[armeabi-v7a] Compile++ thumb: audioengine_static <= AudioPlayerProvider.cpp
make: *** [obj/local/armeabi-v7a/objs/cocos2d_js_static/__/manual/js_manual_conversions.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [obj/local/armeabi-v7a/objs/cocos2d_lua_static/__/auto/lua_cocos2dx_extension_auto.o] Error 1
make: *** [obj/local/armeabi-v7a/objs/cocos2d_js_static/__/auto/jsb_cocos2dx_extension_auto.o] Error 1
Error running command, return code: 2.
Error running command, return code: 14.

**Comment by slackmoehrle** [comment]
v3.17? Are you sure?

I used the v3 branch and compiled the other day. Let me test this out again.

**Comment by deguilardi** [comment]
yes, I am sure

$ cocos -v
Python 2.7.10
cocos2d-x-3.17
Cocos Console 2.3

**Comment by slackmoehrle** [comment]
but 3.17 hasn't been released yet. Must be preparing to release.

I ran this: `$ cocos gen-libs -p android --app-abi armeabi-v7a`

and it was fine:

```
BUILD SUCCESSFUL in 11s
54 actionable tasks: 3 executed, 51 up-to-date
Move apk to /Users/dubb/Cocos/cocos2d-x/tests/cpp-empty-test/bin/debug/android
Build succeed.
```

```
$ cocos -v
Python 2.7.10
cocos2d-x-3.17
Cocos Console 2.3
```

**Comment by deguilardi** [comment]
I re-downloaded everything, restarted the computer and re-setted all the environment. 

Now it is stuck in this error:
Android NDK: ERROR:/Users/deguilardi/cocosWorkspace/lib-cocos2d/external/lua/luajit/prebuilt/android/Android.mk:luajit_static: LOCAL_SRC_FILES points to a missing file    
Android NDK: Check that /Users/deguilardi/cocosWorkspace/lib-cocos2d/external/lua/luajit/prebuilt/android/armeabi/libluajit.a exists  or that its path is correct   
/Users/deguilardi/Library/Android/android-ndk-r13b/build/core/prebuilt-library.mk:45: *** Android NDK: Aborting    .  Stop.

**Comment by deguilardi** [comment]
Check this out!

I ran the command about 20 times, everytime I run it, it goes a little further. It's still crashing the same error, but I guess if I run it 1000x it will compile the intere thing eventually.

**Comment by deguilardi** [comment]
I am running this:
cocos gen-libs -p android --app-abi armeabi-v7a --ap android-16

@slackmoehrle I made a fresh install of the v3.16 and now I'm getting the following error:

[armeabi-v7a] Install        : libcocos2dlua.so => libs/armeabi-v7a/libcocos2dlua.so
Building apk...
only build 32bit
running: '/usr/local/Cellar/ant/1.10.1/libexec/ant clean debug -f /Users/deguilardi/cocosWorkspace/lib-cocos2d/tools/simulator/frameworks/runtime-src/proj.android/build.xml -Dsdk.dir=/Users/deguilardi/Library/Android/sdkv25.2.5'

/bin/sh: /usr/local/Cellar/ant/1.10.1/libexec/ant: No such file or directory
Error running command, return code: 127.
Error running command, return code: 14.

**Comment by deguilardi** [comment]
@slackmoehrle after rolling back to v3.16 the libraries are created with no errors.
The issue now is on Android Studio, the compiler is giving "undefined reference" to every single cocos symbol

Do you have any clue?

**Comment by slackmoehrle** [comment]
can you show me a screenshot?

**Comment by deguilardi** [comment]
@slackmoehrle to help you, the screenshots are showing that all .a libs where generated, but the libcocos2d.a is empty.

all files generated:
![screen shot 2018-02-09 at 9 20 07 am](https://user-images.githubusercontent.com/16661903/36032113-ee27906a-0d7a-11e8-8a43-8fce83cbb7c7.png)

folder size:
![screen shot 2018-02-09 at 9 22 40 am](https://user-images.githubusercontent.com/16661903/36032134-02d99bd4-0d7b-11e8-9ef9-86a5e23648be.png)

libcocos2d.a is empty:
![screen shot 2018-02-09 at 9 21 44 am](https://user-images.githubusercontent.com/16661903/36032147-10e55f56-0d7b-11e8-8af5-9215c78c678f.png)

**Comment by drelaptop** [comment]
what I understand is that all the libs is needed for your project. libcocos2d.a is so small, it's not a error, for the main content is in libcocos2dxinternal.a, every other libs have it's meaning. 

indeed, it's strange, package all the libs into one single lib is more friendly. the same problem occurs to me, the libs that cmake generated is divided too.

**Comment by deguilardi** [comment]
@drelaptop do you know a tutorial to do this s***. Because the official documentation sucks and I am so tired of this.

I am in the point where the libs are generated. All I need to do is import them to my projects.

**Comment by slackmoehrle** [comment]
One thing is you can always run `cocos new ...` and use the result as a skeleton for what you want to do. The Android Studio project it creates will have everything done. Just rename it and start adding your source and resources.

**Comment by deguilardi** [comment]
@slackmoehrle I have more than 30 projects running on cocos. When I have to compile them, it takes me a day and 180 GB on the disk. As my computer is one of those with 256GB SSD, I have to erase all that repeated stuff every time I finish compiling, so I can still use the computer.

All the methods I tried to use don't work, every single tutorial on the web. Tried also downgrading, cloning github, manual build, gradle, ant... nothing works. PS.: Oficial Documentation sucks

**Comment by slackmoehrle** [comment]
Yup, we don't have a lot of documentation about using the pre-built libraries for Android/Android Studio. Tutorials from 3rd party sources are likely to fail now-a-days. Google removing the `android` command-line tool is just one example that breaks. There are several discussions on the forums about using pre-builts for Android. Have you looked at those? I'm sorry that pre-builts on Android are not easy like iOS.

**Comment by slackmoehrle** [comment]
to be sure, you want to run `cocos gen-libs ...` and then use those pre-builts in your Android Studio projects?

**Comment by deguilardi** [comment]
@slackmoehrle yes! This is exactly what I need.

Just by downloading and setting up everything is not enough to generate the libs. I had to open the cpp-empty-test and compile the project by calling the gradle task manually before calling the gen-libs script

Now I can generate the libs, but they are not being split by each arch; The "prebuilt-mk/Android.mk" file is also not being generate by the script. So the pain in the ass is import everything to my studio projects.

**Comment by slackmoehrle** [comment]
I pulled the latest from `v3` and I ran: 
```sh
cocos gen-libs -p android --app-abi armeabi-v7a -p ios -m release
```

and it fails with:
```sh
Move apk to /Users/dubb/Cocos/cocos2d-x/tests/cpp-empty-test/bin/debug/android
Build succeed.
Traceback (most recent call last):
  File "/Users/dubb/Cocos/cocos2d-x/tools/cocos2d-console/bin/cocos.py", line 1072, in <module>
    run_plugin(command, argv, plugins)
  File "/Users/dubb/Cocos/cocos2d-x/tools/cocos2d-console/bin/cocos.py", line 961, in run_plugin
    plugin.run(argv, dependencies_objects)
  File "/Users/dubb/Cocos/cocos2d-x/tools/cocos2d-console/plugins/plugin_generate/gen_libs.py", line 79, in run
    self.compile()
  File "/Users/dubb/Cocos/cocos2d-x/tools/cocos2d-console/plugins/plugin_generate/gen_libs.py", line 176, in compile
    self.compile_android()
  File "/Users/dubb/Cocos/cocos2d-x/tools/cocos2d-console/plugins/plugin_generate/gen_libs.py", line 368, in compile_android
    cocos.copy_files_with_config(copy_cfg, obj_dir, android_out_dir)
  File "/Users/dubb/Cocos/cocos2d-x/tools/cocos2d-console/bin/cocos.py", line 799, in copy_files_with_config
    src_dir, src_dir, dst_dir, include_rules, exclude_rules)
  File "/Users/dubb/Cocos/cocos2d-x/tools/cocos2d-console/bin/cocos.py", line 818, in copy_files_with_rules
    for name in os.listdir(src):
OSError: [Errno 2] No such file or directory: '/Users/dubb/Cocos/cocos2d-x/tests/cpp-empty-test/proj.android/app/build/intermediates/ndkBuild/release/obj/local/armeabi-v7a'
```

**Comment by deguilardi** [comment]
exactly, this is a pain in the ass!! Open the tests/cpp-empty-test project and run the gradle build tasks, so you will have this directory and all files that are missing

**Comment by slackmoehrle** [comment]
Can you share with me anything Cocos related in your `.bash_profile`?

**Comment by deguilardi** [comment]
```
export ANDROID_HOME="/Users/deguilardi/Library/Android/sdk"
export ANDROID_SDK_ROOT=$ANDROID_HOME
export NDK_ROOT="/Users/deguilardi/Library/Android/sdk/ndk-bundle"

export ANT_ROOT=/usr/local/Cellar/ant/1.10.1/libexec/bin
export PATH="$ANT_HOME/bin:$ANDROID_HOME/platform-tools:$COCOS_ROOT:$PATH"

# Add environment variable COCOS_CONSOLE_ROOT for cocos2d-x
export COCOS_CONSOLE_ROOT=/Users/deguilardi/cocosWorkspace/lib-cocos2d/tools/cocos2d-console/bin
export PATH=$COCOS_CONSOLE_ROOT:$PATH

# Add environment variable COCOS_X_ROOT for cocos2d-x
export COCOS_X_ROOT=/Users/deguilardi/cocosWorkspace
export PATH=$COCOS_X_ROOT:$PATH

# Add environment variable COCOS_TEMPLATES_ROOT for cocos2d-x
export COCOS_TEMPLATES_ROOT="/Users/deguilardi/cocosWorkspace/lib-cocos2d/templates"
export PATH=$COCOS_TEMPLATES_ROOT:$PATH
```

**Comment by slackmoehrle** [comment]
You opened up this: `cocos2d-x/tests/cpp-empty-test/proj.android-studio` in Android Studio and executed the build.gradle manually?
<img width="527" alt="screen shot 2018-02-09 at 3 07 19 pm" src="https://user-images.githubusercontent.com/7491758/36047762-fed719a0-0daa-11e8-8104-be7f7bb83031.png">

**Comment by deguilardi** [comment]
yes!

**Comment by slackmoehrle** [comment]
Hmm, ok, thanks. I tried that work and my build still failed. Did you just use the "build" button?

**Comment by deguilardi** [comment]
I double clicked this
![screen shot 2018-02-09 at 3 27 39 pm](https://user-images.githubusercontent.com/16661903/36048604-c7e95cca-0dad-11e8-861b-8e8e81b7f9af.png)


and this:
![screen shot 2018-02-09 at 3 28 07 pm](https://user-images.githubusercontent.com/16661903/36048626-d70c9f0a-0dad-11e8-94eb-74e7d1827e70.png)

**Comment by deguilardi** [comment]
But hitting the build button also works, it installed the app on my phone with no problems

**Comment by slackmoehrle** [comment]
I've got that building now. Shall see the result soon.

Edit: Yup that works for me.

**Comment by deguilardi** [comment]
@slackmoehrle what do you mean? Did you make your game run using the generated libs? Or did you just run the cpp-empty-test app on your phone?

**Comment by slackmoehrle** [comment]
@deguilardi I will continue making the game run using generated libs and see how I make out.

**Comment by drelaptop** [comment]
this issue have been fixed at cocos2d/cocos2d-console#440, and have been updated into cocos2d-x, so close this issue.

the next issue is how to find a proper way to use the generated libs. discuss here #18787
